src/store/indexeddb-local-backend.js

```diff
@@ -292,6 +303,44 @@ LocalIndexedDBStoreBackend.prototype = {
         });
     },
 
+    clearOutOfBandMembers: async function(roomId) {
+        // the approach to delete all members for a room
+        // is to get the min and max state key from the index
+        // for that room, and then delete between those
+        // keys in the store.
+        // this should be way faster than deleting every member
+        // individually for a large room.
+        const readTx = this.db.transaction(
+            ["oob_membership_events"],
+            "readonly");
+        const store = readTx.objectStore("oob_membership_events");
+        const roomIndex = store.index("room");
+        const roomRange = IDBKeyRange.only(roomId);
+
+        const minStateKeyProm = reqAsCursorPromise(
+                roomIndex.openKeyCursor(roomRange, "next"),
+            ).then((cursor) => cursor && cursor.primaryKey[1]);
+        const maxStateKeyProm = reqAsCursorPromise(
+                roomIndex.openKeyCursor(roomRange, "prev"),
+            ).then((cursor) => cursor && cursor.primaryKey[1]);
+        const [minStateKey, maxStateKey] = await Promise.all(
+            [minStateKeyProm, maxStateKeyProm]);
+
+        const writeTx = this.db.transaction(
+            ["oob_membership_events"],
+            "readwrite");
+        const writeStore = writeTx.objectStore("oob_membership_events");
+        const membersKeyRange = IDBKeyRange.bound(
+            [roomId, minStateKey],
+            [roomId, maxStateKey],
+        );
```

Comment:
I'm missing why you can't just delete with `IDBKeyRange.only(roomId)`

Reply:
The primary key of `oob_membership_events` is a composed key of `[roomId, stateKey]`, as you need those two to uniquely identify a member. I didn't try this, but I don't think `IDBKeyRange.only` supports matching only a part of a composed key.

Reply:
Actually, I'm quite certain `IDBKeyRange.only` doesn't support prefix-matches as they are called, on composed keys; I asked this question earlier on stackoverflow to make sure this approach would work across browsers, and the person awnsering said prefix-matches are on peoples wish-lists for idb.

https://stackoverflow.com/questions/52111115/safe-to-assume-same-order-in-index-as-store-with-upper-part-of-key-same/52120855#52120855